spec/03-plugins/11-correlation-id/02-schema_spec.lua

```diff
@@ -0,0 +1,184 @@
+local schema_def = require "kong.plugins.correlation-id.schema"
+local v = require("spec.helpers").validate_plugin_config_schema
+local helpers = require "spec.helpers"
+local uuid = require "resty.jit-uuid"
+local pgmoon_json = require("pgmoon.json")
+local cjson = require "cjson"
+
+describe("Schema: correlation-id", function ()
+  it("requried field must be included", function()
+    local ok, err = v({
+      generator = ngx.null,
+     }, schema_def)
+
+    assert.falsy(ok)
+    assert.is_not_nil(err)
+    assert.equals("required field missing", err.config.generator)
+  end)
+end)
+
+local strategy = "postgres"
+describe("Plugin: correlation-id (schema) #a [#" .. strategy .."]", function()
+  local admin_client, bp, db, plugin_id,ws
+  local plugin_config = {
+    generator = ngx.null,
+    header_name = "Kong-Request-ID",
+    echo_downstream = true,
+  }
+
+  local function render(template, keys)
+    return (template:gsub("$%(([A-Z_]+)%)", keys))
+  end
+
+  lazy_setup(function()
+    local plugin_name = "correlation-id"
+    bp, db = helpers.get_db_utils(strategy, { "plugins", "workspaces", })
+    ws = db.workspaces:select_by_name("default")
+    assert.is_truthy(ws)
+    plugin_id = uuid.generate_v4()
+    local sql = render([[
+      INSERT INTO plugins (id, name, config, enabled, ws_id) VALUES
+        ('$(ID)', '$(PLUGIN_NAME)', $(CONFIG)::jsonb, TRUE, '$(WS_ID)');
+      COMMIT;
+    ]], {
+      ID = plugin_id,
+      PLUGIN_NAME = plugin_name,
+      CONFIG = pgmoon_json.encode_json(plugin_config),
+      WS_ID = ws.id,
+    })
+
+    local res, err = db.connector:query(sql)
```

Comment:
Just curious why the plugin is inserted directly with a sql query instead of using the high level `bp`.

Reply:
The purpose of the this test case is to verify even if the generator field in database is `null`, the plugin object can still be constructed successfully, the field can get its default value when `required = true` is attached in the schema.
To setup such record in database, we have to use raw SQL since it will fail with `bp` due to the change of the schema.